test/test_enzyme.jl

```diff
@@ -1,13 +1,78 @@
 using Oceananigans
+using Oceananigans.TurbulenceClosures: with_tracers
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Fields: ConstantField
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: tracernames
 using Enzyme
 
 # Required presently
 Enzyme.API.runtimeActivity!(true)
-
-EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.API.looseTypeAnalysis!(true)
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Clock}) = true
+Enzyme.EnzymeRules.inactive_noinl(::typeof(Core._compute_sparams), args...) = nothing
 
 f(grid) = CenterField(grid)
 
+const maximum_diffusivity = 100
+
+"""
+    set_diffusivity!(model, diffusivity)
+
+Change diffusivity of model to `diffusivity`.
+"""
+function set_diffusivity!(model, diffusivity)
+    closure = VerticalScalarDiffusivity(; κ=diffusivity)
+    names = tuple(:c) # tracernames(model.tracers)
+    closure = with_tracers(names, closure)
+    model.closure = closure
+    return nothing
+end
+
+function set_initial_condition!(model, amplitude)
+    # Set initial condition
+    amplitude = Ref(amplitude)
+
+    # This has a "width" of 0.1
+    cᵢ(x, y, z) = amplitude[] * exp(-z^2 / 0.02 - (x^2 + y^2) / 0.05)
+    set!(model, c=cᵢ)
+
+    return nothing
+end
+
+function stable_diffusion!(model, amplitude, diffusivity)
+    set_diffusivity!(model, diffusivity)
+    set_initial_condition!(model, amplitude)
+    
+    # Do time-stepping
+    Nx, Ny, Nz = size(model.grid)
+    κ_max = maximum_diffusivity
+    Δz = 2π / Nz
+    Δt = 1e-1 * Δz^2 / κ_max
+
+    model.clock.time = 0
+    model.clock.iteration = 0
+
+    for n = 1:10
+        time_step!(model, Δt; euler=true)
+    end
+
+    # Compute scalar metric
+    c = model.tracers.c
+
+    # Hard way
+    # c² = c^2
+    # sum_c² = sum(c²)
+
+    # Another way to compute it
+    sum_c² = 0.0
+    for i = 1:Nx, j = 1:Ny, k = 1:Nz
+        sum_c² += c[i, j, k]^2
+    end
+
+    return sum_c²::Float64
```

Comment:
```suggestion
    return sum_c²
```

Reply:
for some reason that may be important for type inference and thus auto-diffing but I'm not sure

Reply:
Yeah otherwise Julia cannot infer the return type (which presents issues here).

Reply:
OK, perhaps add a comment?

Reply:
Added